src/twisted/test/test_twistd.py

```diff
@@ -865,6 +901,91 @@ def fakeSetupEnvironment(self, chroot, rundir, nodaemon, umask,
             ['/foo/chroot', '/foo/rundir', True, 56, '/foo/pidfile'])
 
 
+    def test_shedPrivileges(self):
+        """
+        """
+        def switchUIDPass(euid, uid, gid):
+            """
+            Ignore everything and pass
+            """
+        self.patch(_twistd_unix, 'switchUID', switchUIDPass)
+        runner = UnixApplicationRunner({})
+        runner.shedPrivileges(35, 200, None)
+
+
+    def test_shedPrivilegesError(self):
+        """
+        An unexpected L{OSError} when calling
+        L{twisted.scripts._twistd_unix.shedPrivileges}
+        terminates the process via L{SystemExit}.
+        """
+        def switchUIDFail(euid, uid, gid):
+            raise OSError(errno.EBADF, "fake")
+
+        runner = UnixApplicationRunner({})
+        self.patch(_twistd_unix, 'switchUID', switchUIDFail)
+        exc = self.assertRaises(SystemExit, runner.shedPrivileges, 35,
+                                200, None)
+        self.assertEqual(exc.code, 1)
+
+
+    def _setUID(self, wantedUser, wantedUid, wantedGroup, wantedGid):
+        """
+        Common code for tests which try to pass the the UID to
+        L{UnixApplicationRunner}.
+        """
+        patchUserDatabase(self.patch, wantedUser, wantedUid, wantedGroup,
+                          wantedGid)
+
+        def initgroups(uid, gid):
+            self.assertEqual(uid, wantedUid)
+            self.assertEqual(gid, wantedGid)
+
+        def setuid(uid):
+            self.assertEqual(uid, wantedUid)
+
+        def setgid(gid):
+            self.assertEqual(gid, wantedGid)
+
+        from twisted.python import util
```

Comment:
Why not put this import at the top of the file?

Reply:
Fixed